opentelemetry/src/main/java/io/grpc/opentelemetry/OpenTelemetryModule.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2023 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.opentelemetry;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.grpc.internal.GrpcUtil.IMPLEMENTATION_VERSION;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Supplier;
+import io.grpc.ClientInterceptor;
+import io.grpc.ExperimentalApi;
+import io.grpc.ServerStreamTracer;
+import io.grpc.opentelemetry.internal.OpenTelemetryConstants;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.MeterProvider;
+
+
+@ExperimentalApi("https://github.com/grpc/grpc-java/issues/10591")
+public final class OpenTelemetryModule {
+
+  private static final Supplier<Stopwatch> STOPWATCH_SUPPLIER = new Supplier<Stopwatch>() {
+    @Override
+    public Stopwatch get() {
+      return Stopwatch.createUnstarted();
+    }
+  };
+
+  private final OpenTelemetry openTelemetryInstance;
+  private final MeterProvider meterProvider;
+  private final Meter meter;
+  private final OpenTelemetryMetricsState state;
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  private OpenTelemetryModule(Builder builder) {
+    this.openTelemetryInstance = checkNotNull(builder.openTelemetrySdk, "openTelemetrySdk");
+    this.meterProvider =
+        openTelemetryInstance.getMeterProvider() != null ? openTelemetryInstance.getMeterProvider()
```

Comment:
Added it thinking user's implementation might return null. After discussing offline, it makes sense that even if it return null since it is going to raise an exception early on we do not need this check here. Removed.